pkg/common/util.go

```diff
@@ -87,6 +87,40 @@ func ToLowerCaseKeys(keyValues map[string]string) map[string]string {
 	return newMap
 }
 
+// Recursively truncates the given "field" from CustomResourceDefinition to 50 characters.
+func truncateFieldRecursively(data map[string]interface{}, maxLength int, field string) {
+	for key, value := range data {
+		if key == field {
+			if str, ok := value.(string); ok && len(str) > maxLength {
+				data[key] = str[:maxLength]
+			}
+			continue
+		}
+		if subObj, ok := value.(map[string]interface{}); ok {
+			truncateFieldRecursively(subObj, 50, field)
+		}
+		if subObjs, ok := value.([]interface{}); ok {
+			for _, subObj := range subObjs {
+				if subObjMap, ok := subObj.(map[string]interface{}); ok {
+					truncateFieldRecursively(subObjMap, 50, field)
+				}
+			}
+		}
+	}
+}
+
+// truncates the given "field" from CustomResourceDefinition to 50 characters.
+func TruncateField(field string, targetLength int) manifestival.Transformer {
```

Comment:
Since this has special behavior for CRDs and implements the `Transformer` interface, I recommend updating the function name to make this more clear. I also updated the length variable to `maxLength` so it is consistent.

```suggestion
// TruncateCRDFieldTransformer returns a manifestival.Transformer that truncates the value of the given field within a CRD spec to the provided max length.
func TruncateCRDFieldTransformer(field string, maxLength int) manifestival.Transformer {
```